core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/DataProcessor.scala

```diff
@@ -0,0 +1,146 @@
+package edu.uci.ics.amber.engine.architecture.worker.neo
+
+import java.util.concurrent.{ExecutorService, Executors, Future}
+
+import akka.actor.{Actor, ActorRef}
+import edu.uci.ics.amber.engine.architecture.breakpoint.localbreakpoint.ExceptionBreakpoint
+import edu.uci.ics.amber.engine.architecture.worker.BreakpointSupport
+import edu.uci.ics.amber.engine.common.amberexception.BreakpointException
+import edu.uci.ics.amber.engine.common.ambermessage.ControlMessage.LocalBreakpointTriggered
+import edu.uci.ics.amber.engine.common.ambermessage.WorkerMessage.ExecutionCompleted
+import edu.uci.ics.amber.engine.common.{IOperatorExecutor, InputExhausted}
+import edu.uci.ics.amber.engine.common.tuple.ITuple
+
+
+class DataProcessor(                             // dependencies:
+                     operator: IOperatorExecutor,// core logic
+                     tupleInput: TupleInput,     // to get input tuples
+                     tupleOutput: TupleOutput,   // to send output tuples
+                     pauseUtil: PauseUtil,       // to pause/resume
+                     self:ActorRef               // to notify main actor
+                   ) extends BreakpointSupport { // TODO: make breakpointSupport as a module
+
+  private[this] def funToRunnable(fun: () => Unit): Runnable = new Runnable() { def run(): Unit = fun() }
```

Comment:
remove this function. You can put it all in dpThreadFuture. Code will be actually more readable that way

Reply:
Similarly you don't need to define executorService. Directly use `Executors.newSingleThreadExecutor`